request-attrs.go

```diff
@@ -44,6 +44,37 @@ func newFileAttrFlags(flags uint32) FileAttrFlags {
 	}
 }
 
+func (r *Request) SetAttributes(flags FileAttrFlags, attrs FileStat) error {
```

Comment:
The current name isn't the most obvious, though it does match the Attributes call which produces a FileStat from the Flags and Attrs of a Request object.

How would you feel about a NewRequest variant for this case?  I'm not entirely sold on it, but I do think that it would be a mildly better fit than a Marshal on FileStat.

Because what we want out of this really isn't a []byte, it's a Request object that is suitable for Setattr.  Unfortunately, with Request being a mixture of marshaled and unmarshaled data that is easier said than done.


As an example, my external code that uses h.FilePut.Filewrite does a reasonably horrible:

```
    r := sftp.NewRequest("Put", name)
    // These are SSH flags, which, naturally, are not exported.
    // This is write, create, exclusive.
    r.Flags = 2 | 8 | 0x20
    
    w, err := h.FilePut.Filewrite(r)
    if err != nil {
        log.Fatalf("Unable to get filewrite for %s: %v", name, err)
    }
```

And that is specifically because we have no inverse of newFileOpenFlags which is exported.

Ideally we would just use os.OpenFile flags for the flags in a Request, but again, that would be a potentially breaking change.

It would be nice to come up with a pattern that worked well enough to make it easy to construct a proper Request in general, for the purpose of populating things like memfile in server code.

The other option I can see is making a variant of Request that has everything fully unmarshaled, and having a function to convert it to Request, and another to convert Request to it.

That frankly feels a little better, and I can see it being more friendly in the long term to convert to the 'friendly' version inside things like memfile, instead of calling things like Pflags, AttrFlags, FileMode, and Attributes on Request every time we need those values parsed out.

I may try and write that out, just to see what it looks like.